module/experiment.js

```diff
@@ -2,27 +2,46 @@ import React from 'react'
 import PropTypes from 'prop-types'
 import { connect } from 'react-redux'
 import { Variant } from './variant'
-import { setExperimentVariant } from './actions'
+import { resetExperimentVariant } from './actions'
 
 export function Selector (props) {
-	const { variant, name, children } = props
-	let chosenOne
-	React.Children.forEach(children, child => {
-		if (child.type === Variant && child.props.name === variant) {
-			chosenOne = React.cloneElement(child, { experiment: name })
-		}
-	})
+    const { variant, name, children, resetExperimentVariant } = props
+    const variants = getComponentVariants(children)
+    let chosenOne
+    if (variant === null) {
+        resetExperimentVariant(name, variants)
+    } else {
+        React.Children.forEach(children, child => {
+            if (child.type === Variant && child.props.name === variant) {
+                chosenOne = React.cloneElement(child, { experiment: name })
+            }
+        })
+        if (chosenOne === undefined && variants.length > 0) {
+              resetExperimentVariant(name, variants)
+        }
+    }
 	return chosenOne || null
 }
 
 Selector.propTypes = {
 	name: PropTypes.string.isRequired,
 }
 
+function getComponentVariants(children) {
+    return React.Children.map(children, child => child.type === Variant
+        ? { name: child.props.name } : null).filter(variant => !!variant)
+}
+
 const mapStateToProps = (state, ownProps) => {
 	return {
 		variant: state.experiments[ownProps.name] || null
 	}
 }
 
-export const Experiment = connect(mapStateToProps)(Selector)
\ No newline at end of file
+const mapDispatchToProps = dispatch => {
+    return {
+        resetExperimentVariant: (experiment, variants) => dispatch(resetExperimentVariant(experiment, variants))
+    }
+}
+
+export const Experiment = connect(mapStateToProps, mapDispatchToProps)(Selector)
```

Comment:
you could inline mapStateToProps and mapDispatchToProps, it would result in less code